sdk/storage/Azure.Storage.Blobs.ChangeFeed/src/Segment.cs

```diff
@@ -0,0 +1,195 @@
+                jsonManifest = await JsonDocument.ParseAsync(blobDownloadInfo.Content).ConfigureAwait(false);
+            }
+            else
+            {
+                jsonManifest = JsonDocument.Parse(blobDownloadInfo.Content);
+            }
+
+            // Initalized Finalized field
+            string statusString = jsonManifest.RootElement.GetProperty("status").GetString();
+            Finalized = statusString == "Finalized";
+
+            int i = 0;
+            foreach (JsonElement shardJsonElement in jsonManifest.RootElement.GetProperty("chunkFilePaths").EnumerateArray())
+            {
+                //TODO cleanup this line
+                string shardPath = shardJsonElement.ToString().Substring("$blobchangefeed/".Length);
+                Shard shard = new Shard(_containerClient, shardPath, _cursor?.ShardCursors?[i]);
+                _shards.Add(shard);
+                i++;
+            }
+            _isInitalized = true;
+        }
+
+        public SegmentCursor GetCursor()
+        {
+            List<ShardCursor> shardCursors = new List<ShardCursor>();
+            foreach (Shard shard in _shards)
+            {
+                shardCursors.Add(shard.GetCursor());
+            }
+            return new SegmentCursor(
+                segmentDateTime: DateTime,
+                shardCursors: shardCursors,
+                shardIndex: _shardIndex);
+        }
+
+        public async Task<List<BlobChangeFeedEvent>> GetPage(
+            bool async,
+            int? pageSize,
+            CancellationToken cancellationToken = default)
+        {
+            List<BlobChangeFeedEvent> changeFeedEventList = new List<BlobChangeFeedEvent>();
+
+            if (!_isInitalized)
+            {
+                await Initalize(async).ConfigureAwait(false);
+            }
+
+            if (!HasNext())
+            {
+                throw new InvalidOperationException("Segment doesn't have any more events");
+            }
+
+            int i = 0;
+            while (i < pageSize && _shards.Count > 0)
+            {
+                Shard currentShard = _shards[_shardIndex];
+
+                BlobChangeFeedEvent changeFeedEvent = await currentShard.Next(async, cancellationToken).ConfigureAwait(false);
+
+                changeFeedEventList.Add(changeFeedEvent);
+
+                // If the current shard is completed, remove it from _shards
+                if (!currentShard.HasNext())
+                {
+                    _shards.RemoveAt(_shardIndex);
+                }
+
+                i++;
+                _shardIndex++;
+                if (_shardIndex >= _shards.Count)
+                {
+                    _shardIndex = 0;
+                }
+            }
+
+            //TODO how to get raw response for page?  Does it matter?
+            return changeFeedEventList;
+        }
+
+        //TODO figure out if this is right.
+        public bool HasNext()
+        {
+            if (!_isInitalized)
```

Comment:
Should this trigger initialize instead? or throw? Good test for this assumption might be - fetch change log for account with zero activity.

Reply:
Now we that have factories, _isInitalized isn't a thing anymore in ChangeFeed, Segment, or Shard.